test/kitchen/.kitchen-azure.yml

```diff
@@ -35,25 +35,12 @@ provisioner:
 platforms:
 # Loop through two lists and output a total matrix of all possible platform + chef versions,
 # for both drivers
-<% azure_test_platforms = [
-      # working
-      ['centos-74', 'OpenLogic:CentOS:7.4:7.4.20171110'],
-      ['ubuntu-14-04', 'Canonical:UbuntuServer:14.04.5-LTS:14.04.201711151'],
-      ['ubuntu-16-04','Canonical:UbuntuServer:16.04.0-LTS:16.04.201610200'],
-      ['debian-8', 'credativ:Debian:8:8.0.201801170'],
-      ['sles-12', 'SUSE:SLES:12-SP3:2017.12.11'],
 
-      ['win2012', 'MicrosoftWindowsServer:WindowsServer:2012-Datacenter:3.127.20171115'],
-      ['win2012r2', 'MicrosoftWindowsServer:WindowsServer:2012-R2-Datacenter:4.127.20171115'],
-      ['win2016', 'MicrosoftWindowsServer:WindowsServer:2016-Datacenter:2016.127.20171116'],
 
-      # broken:
-      # ['win2008r2', 'MicrosoftWindowsServer:WindowsServer:2008-R2-SP1:2.127.20171115'],
-      # ['centos-69', 'OpenLogic:CentOS:6.9:6.9.20170612'],
-      # ['debian-7', 'credativ:Debian:7:7.0.201701180'],
-      # This cannot be enabled until dirmngr issue is resolved
-      # ['debian-9', 'credativ:Debian:9:9.0.201710090']
-    ]
+<%
+    # TEST_PLATFORMS syntax is `short_name1,azure_full_qualified_name1|short_name2,azure_full_qualified_name1`
+    azure_test_platforms = ENV['TEST_PLATFORMS'].split('|').map { |p| p.split(',') }
```

Comment:
ideally I think the available platforms should still be listed in this file. The `kitchen` command support providing a regex of the test suite(s) to run for a given `kitchen test` invocation, so this could be used to run `kitchen test` on the desired platforms.

Let's merge as-is though, if you have the time feel free to address this in a separate PR.